pkg/apiclient/offline-client.go

```diff
@@ -50,42 +49,43 @@ func newOfflineClient(paths []string) (context.Context, Client, error) {
 
 	for _, basePath := range paths {
 		err := file.WalkManifests(basePath, func(path string, bytes []byte) error {
-			var generic map[string]interface{}
-			if err := yaml.Unmarshal(bytes, &generic); err != nil {
-				return fmt.Errorf("failed to parse YAML from file %s: %w", path, err)
-			}
-			switch generic["kind"] {
-			case "ClusterWorkflowTemplate":
-				cwftmpl := new(wfv1.ClusterWorkflowTemplate)
-				if err := yaml.Unmarshal(bytes, &cwftmpl); err != nil {
-					return fmt.Errorf("failed to unmarshal file %s as a ClusterWorkflowTemplate: %w", path, err)
+			for _, pr := range common.ParseObjects(bytes, false) {
+				obj, err := pr.Object, pr.Err
+				if err != nil {
+					return fmt.Errorf("failed to parse YAML from file %s: %w", path, err)
 				}
 
-				if _, ok := clusterWorkflowTemplateGetter.clusterWorkflowTemplates[cwftmpl.Name]; ok {
-					return fmt.Errorf("duplicate ClusterWorkflowTemplate found: %q", cwftmpl.Name)
+				if obj == nil {
+					continue // could not parse to kubernetes object
```

Comment:
This too is copied from `lint.go`, although I would think both places should have an error in this case rather than a silent `continue`.